nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-processors/src/main/java/org/apache/nifi/processors/kafka/GetKafka.java

```diff
@@ -297,6 +320,14 @@ protected PropertyDescriptor getSupportedDynamicPropertyDescriptor(final String
                 .build();
     }
 
+    @OnScheduled
+    public void schedule(ProcessContext context) {
+        this.deadlockTimeout = context.getProperty(KAFKA_TIMEOUT).asTimePeriod(TimeUnit.MILLISECONDS) * 2;
+        if (this.executor == null || this.executor.isShutdown()) {
+            this.executor = Executors.newFixedThreadPool(Runtime.getRuntime().availableProcessors());
```

Comment:
Think instead it should use newCachedThreadPool .  Behavior seems ideal for this case.  Need this in both procs.